crates/optimism/consensus/src/lib.rs

```diff
@@ -99,13 +104,42 @@ impl<ChainSpec: EthChainSpec + OpHardforks, B: Block> Consensus<B>
             return Err(ConsensusError::BodyTransactionRootDiff(error.into()))
         }
 
-        // EIP-4895: Beacon chain push withdrawals as operations
-        if self.chain_spec.is_shanghai_active_at_timestamp(block.timestamp()) {
-            validate_shanghai_withdrawals(block)?;
+        // Check empty shanghai-withdrawals
+        if self.chain_spec.is_canyon_active_at_timestamp(block.timestamp()) {
+            canyon::verify_empty_shanghai_withdrawals(block.body()).map_err(|err| {
+                trace!(target: "op::consensus",
+                    block_number=block.number(),
+                    %err,
+                    "block failed validation",
+                );
+
+                ConsensusError::Other
+            })?;
+        } else {
+            return Ok(())
         }
 
         if self.chain_spec.is_cancun_active_at_timestamp(block.timestamp()) {
             validate_cancun_gas(block)?;
+        } else {
+            return Ok(())
+        }
+
+        // Check withdrawals root field in header
+        if self.chain_spec.is_isthmus_active_at_timestamp(block.timestamp()) {
```

Comment:
shouldn't this check be applied before the canyon checks?

Reply:
from canyon to isthmus: withdrawals field should be empty root hash
from isthmus on: withdrawals field should _not_ be empty root hash

so is required to check isthmus first, yes